### DISCUSSION TITLE
Grid view not showing heavy nodes

### DISCUSSION THREAD

**Discussion by petiepooo** [Security-Onion-Solutions/securityonion#8580] [Unsupported Versions] [answered]
I'm testing various grid configurations and running into an issue I cannot figure out.

I have a ManagerSearch node on v2.3.140 with two HeavyNodes.  Cross-cluster search is working, as I see events from all three in Alerts and Kibana.  Grafana is showing performance info on all three.  Unfortunately, the grid view in SOC is only showing the manager node.

It appears the SOC software is responsible for rendering the list of nodes.  Where does that info come from, and how can I debug it?

Note: I installed as ManagerSearch instead of just Manager to avoid the redis queue draining issue.  I understand heavy nodes are only for low-throughput environments, and this qualifies.  I would normally run two standalone nodes, but wanted the cross-cluster search capability.

For reference, I also have a Standalone node on 2.3.140 with two ForwardNodes, and everything looks fine.  I see events from all three in Kibana, Grafana shows performance data for all, and the grid view in SOC shows all three nodes.  I can use it as a reference to view where the data for the grid view comes from and is rendered, but my searches for how that's done came up empty.

Thank you.

**Comment by dougburks** [comment, marked as answer]
Works for me:

<img width="1659" alt="ManagerSearch_and_Heavy_Node_SOC_grid" src="https://user-images.githubusercontent.com/1659467/186464396-d45f3c3d-9e9d-4d3b-a33f-5ef7feb1c473.png">

Make sure that the heavy nodes can connect to the manager using the ports listed here:
https://docs.securityonion.net/en/2.3/firewall.html#node-communication

**Reply by petiepooo** [reply]
Yup, that was it... sort of.

For the curious, this is the summary I was looking for:  The sensoroni container on all nodes connects to the manager node at the URL specified in the node's /opt/so/conf/sensoroni/sensoroni.json file.  That connection populates the SOC container's data store with the nodes' parameters and pushes periodic updates.  If a node is not populating, it means sensoroni is not pushing updates.

In this case, because the manager's main URL was specified using the IP on one network (from where analysts will be connecting) but the heavy nodes connect to the manager via a different network (a wireguard link), there was no route from the heavy nodes to the manager's main URL's IP.  Everything else (salt, docker, and telegraf) worked fine as they use the hostname given during the heavy node setup, which I set to the manager's wireguard IP.  A change to the wireguard config to route the main URL IP via its link was enough to get sensoroni working too.

Thanks!
